src/connection.ts

```diff
@@ -290,6 +293,17 @@ interface AzureActiveDirectoryServicePrincipalSecret {
   };
 }
 
+/** Structure that defines the options that are necessary to authenticate the tedious instance with an `@azure/identity` chained token credential derived class. */
+interface CredentialChainAuthentication {
+  /** Unique designator for the type of authentication to be used. */
+  type: 'microsoft-credential-chain';
+  /** Set of configurations that are required or allowed with this authentication type. */
+  options: {
+    /** Credential object used to authenticate to the resource. */
+    credential: ChainedTokenCredential;
```

Comment:
I think we should change the type here to `credential: TokenCredential | (): TokenCredential | (): Promise<TokenCredential>`

This would allow clients to re-use the connection configuration and re-generate tokens whenever a connection is opened with the same configuration automatically.

Reply:
Also, since `Azure AD` is the old name of Microsoft's directory service. I think `microsoft-token-credential` would be a better name since it would be future proofed for when MSFT eventually renames the or changes the terminology.

What do you think?

Reply:
> I think we should change the type here to `credential: TokenCredential | () => TokenCredential | () => Promise<TokenCredential>`
> 
> This would allow clients to re-use the connection configuration and re-generate tokens whenever a connection is opened with the same configuration automatically.

Also, did I read this right? do you want me to create some sort of function invoker thingy?

Reply:
Yeah, but we can also just skip this for now and add it later. Sorry for the confusion 🙇‍♂️

Reply:
I will make a second Issue for this for this behavior.